src/manager/monitor.c

```diff
@@ -194,6 +195,84 @@ static int monitor_method_close(sd_bus_message *m, void *userdata, UNUSED sd_bus
  **************** org.containers.hirte.Monitor.Subscribe *****************
  ************************************************************************/
 
+static bool subscribe_on_node(Manager *manager, Monitor *monitor, const char *node, const char *unit) {
+        _cleanup_free_ char *unit_copy = NULL;
+        if (!copy_str(&unit_copy, unit)) {
+                return false;
+        }
+
+        char *saveptr = NULL;
+        char *unit_name = strtok_r(unit_copy, ",", &saveptr);
+        while (unit_name != NULL) {
+                _cleanup_subscription_ Subscription *sub = create_monitor_subscription(
+                                monitor, node, unit_name);
+                if (sub == NULL) {
+                        return false;
+                }
+
+                /* wildcard subscriptions on units are managed by the respective node */
+                if (!streq(sub->unit, "*")) {
+                        LIST_APPEND(subscriptions, monitor->subscriptions, subscription_ref(sub));
+                }
+                manager_add_subscription(manager, sub);
+
+                unit_name = strtok_r(NULL, ",", &saveptr);
+        }
+
+        return true;
+}
+
+static bool unsubscribe_on_nodes(Manager *manager, Monitor *monitor, const char *nodes, const char *unit);
+static bool subscribe_on_nodes(Manager *manager, Monitor *monitor, const char *nodes, const char *unit) {
+        bool success = true;
+
+        _cleanup_free_ char *nodes_copy = NULL;
+        if (!copy_str(&nodes_copy, nodes)) {
+                return false;
+        }
+
+        char *saveptr = NULL;
+        char *node_name = strtok_r(nodes_copy, ",", &saveptr);
+        while (node_name != NULL) {
+                if (!subscribe_on_node(manager, monitor, node_name, unit)) {
+                        success = false;
+                        break;
+                }
+
+                node_name = strtok_r(NULL, ",", &saveptr);
+        }
+
+        // all or none
+        // cleanup already created subscriptions if one fails to prevent dangling subscriptions
+        if (!success) {
+                unsubscribe_on_nodes(manager, monitor, nodes, unit);
+        }
+
+        return success;
+}
+
+static bool unsubscribe_on_all(Manager *manager, Monitor *monitor, const char *unit);
+static bool subscribe_on_all(Manager *manager, Monitor *monitor, const char *unit) {
+        bool success = true;
+
+        Node *current_node = NULL;
+        Node *next_node = NULL;
+        LIST_FOREACH_SAFE(nodes, current_node, next_node, manager->nodes) {
+                if (!subscribe_on_node(manager, monitor, current_node->name, unit)) {
+                        success = false;
+                        break;
+                }
+        }
```

Comment:
There was already support in manager.c to handle node == NULL in subscribe meaning to match for all nodes. However, you removed that and replaced it with this loop instead. I don't really think this approach is better. You will just get more Subscription objects, and you need the cleanup unsubscribe below that wasn't otherwise needed.

Is there a pro to having one Subscription per node in this case?